### ISSUE TITLE
PDF Protection

### ISSUE THREAD

**Issue by ntthanh** [sgr-ksmt/PDFGenerator#33]
Could you please add password protection for your lib?

**Comment by sgr-ksmt** [comment]
@ntthanh thanks for posting issue!
I will implement this feature. please wait for a while 🙇

**Comment by ntthanh** [comment]
Thanks very much!

Have a good day and hope the feature will be in implemented soon :-)

Best,
Thanh

-----Original Message-----

**Comment by sgr-ksmt** [comment]
@ntthanh 
I apologize for making you wait a long time... 🙇 
I just have implemented password-protection in [v1.3.0](https://github.com/sgr-ksmt/PDFGenerator/releases/tag/1.3.0).

example is following: 

#### Usage

```swift
do {
    // password protection (with string)
    try PDFGenerator.generate([v1, v2], outputPath: "path/to/file.pdf", password: "123abc")
    // or use PDFPassword model
    try PDFGenerator.generate([v1, v2], outputPath: "path/to/file.pdf", password: PDFPassword("123abc"))
    // or use PDFPassword model and set user/owner password.
    try PDFGenerator.generate([v1, v2], outputPath: "path/to/file.pdf", password: PDFPassword(user: "123abc", owner: "ABC123")
} catch PDFGenerateError.InvalidPassword(let password) {
    // ... if password can not convert ASCII.
} catch PDFGenerateError.TooLongPassword(let length) {
    // ... if password length is over 32.
}
```

**Comment by ntthanh** [comment]
Thank you very much.  I will try this feature soon.  Have a good day and
thanks for your great works.

Best,
Thanh
